Demo/HelloWorld/Client/src/Program.cs

```diff
@@ -0,0 +1,212 @@
+                                                                                                                   .ToList();
+
+                                                                                     // Add the result Ids
+                                                                                     resultIds.AddRange(taskResultIfs);
+
+                                                                                     // Task request with payload for the task
+                                                                                     // Also contains the list of results that will be created by the task
+                                                                                     return new TaskRequest
+                                                                                            {
+                                                                                              ExpectedOutputKeys =
+                                                                                              {
+                                                                                                taskResultIfs,
+                                                                                              },
+                                                                                              // Avoid unnecessary copy but data could be modified during sending if the
+                                                                                              // reference to the object is still available
+                                                                                              // This is not the case here
+                                                                                              Payload = UnsafeByteOperations.UnsafeWrap(Encoding.ASCII.GetBytes(input)),
+                                                                                            };
+                                                                                   }))
+                                               .ConfigureAwait(false);
+
+    WriteLine($"sessionId: {createSessionReply.SessionId}");
+
+    // Iterate over result Ids to wait for them and print their value
+    foreach (var resultId in resultIds)
+    {
+      // Result request that the describes the result we want
+      var resultRequest = new ResultRequest
+                          {
+                            ResultId = resultId,
+                            Session  = createSessionReply.SessionId,
+                          };
+
+      // Blocking call that waits for the availability of the result
+      var availabilityReply = submitterClient.WaitForAvailability(resultRequest);
+
+      // Process the result according the the result of the wait
+      switch (availabilityReply.TypeCase)
```

Comment:
Not directly related to PR, but I think it could be nice to have a method of the reply that does this switch under the hood, like `ThrowIfError` of `EnsureOk`.

Reply:
We also need it for several other methods such as the session creation and the waitfor*.